File: src/test/java/leets/leenk/domain/feed/domain/service/FeedDeleteServiceTest.java
package leets.leenk.domain.feed.domain.service;

import leets.leenk.domain.feed.domain.entity.Feed;
import leets.leenk.domain.feed.test.FeedTestFixture;
import leets.leenk.domain.feed.test.UserTestFixture;
import leets.leenk.domain.user.domain.entity.User;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import java.time.LocalDateTime;

import static org.assertj.core.api.Assertions.assertThat;

public class FeedDeleteServiceTest {
    private final FeedDeleteService feedDeleteService = new FeedDeleteService();

    @Test
    @DisplayName("feedDelete 테스트")
    void feedDelete() {
        // given
        User user = UserTestFixture.createUser(1L, "me");
        Feed feed = FeedTestFixture.createFeed(1L, user);

        // when
        LocalDateTime start = LocalDateTime.now();

        feedDeleteService.delete(feed);

        LocalDateTime end = LocalDateTime.now();

        // then
        assertThat(feed.getDeletedAt()).isNotNull();
        assertThat(feed.getDeletedAt()).isAfterOrEqualTo(start);
        assertThat(feed.getDeletedAt()).isBeforeOrEqualTo(end);
    }
}

File: src/test/kotlin/leets/leenk/domain/feed/domain/service/FeedDeleteServiceTest.kt
package leets.leenk.domain.feed.domain.service

import io.kotest.core.spec.style.BehaviorSpec
import io.kotest.matchers.shouldBe
import io.kotest.matchers.shouldNotBe
import leets.leenk.domain.feed.test.FeedTestFixture
import leets.leenk.domain.feed.test.UserTestFixture
import java.time.LocalDateTime

class FeedDeleteServiceTest :
    BehaviorSpec({
        val feedDeleteService = FeedDeleteService()

        Given("사용자와 피드가 주어졌을 때") {
            val user = UserTestFixture.createUser(1L, "me")
            val feed = FeedTestFixture.createFeed(1L, user)

            When("피드를 삭제하면") {
                val start = LocalDateTime.now()
                feedDeleteService.delete(feed)
                val end = LocalDateTime.now()

                Then("deletedAt이 현재 시간으로 설정되어야 한다") {
                    feed.deletedAt shouldNotBe null
                    (feed.deletedAt!! >= start) shouldBe true
                    (feed.deletedAt!! <= end) shouldBe true
                }
            }
        }
    })
